go-controller/pkg/allocator/ip/subnet/allocator_test.go

```diff
@@ -216,6 +216,64 @@ var _ = ginkgo.Describe("Subnet IP allocator operations", func() {
 
 	})
 
+	// Reserved subnets test cases
+	ginkgo.It("reserves subnets correctly and allows specific allocation", func() {
+		subnets := []string{
+			"10.1.1.0/24",
+		}
+		reservedSubnets := []string{
+			"10.1.1.0/28", // Reserve 10.1.1.0-15
+		}
+		expectedIP := "10.1.1.16/24"
+
+		err := allocator.AddOrUpdateSubnet(subnetName, ovntest.MustParseIPNets(subnets...), ovntest.MustParseIPNets(reservedSubnets...))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ips, err := allocator.AllocateNextIPs(subnetName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(ips).To(gomega.HaveLen(1))
+		gomega.Expect(ips[0].String()).To(gomega.Equal(expectedIP))
+
+		// Should be able to allocate specific IPs from reserved range
+		reservedIPs, err := util.ParseIPNets([]string{"10.1.1.5/24"})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = allocator.AllocateIPPerSubnet(subnetName, reservedIPs)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
nit you can ignore:

IMHO, it is better to inline these, since it allows for positive logic:
```suggestion
		gomega.Expect(allocator.AllocateIPPerSubnet(subnetName, reservedIPs)).To(gomega.Succeed())
```

Reply:
Plus, is there a way for us to know that the IP allocation really succeeded ? e.g. allocate the same IP again and see it returns an error - or something like that. 

If so, I'd say we should exercise it.